### DISCUSSION TITLE
Is it safe to pipe age directly in to tar?

### DISCUSSION THREAD

**Discussion by older-pack** [FiloSottile/age#599] [Q&A] [answered]
Is this workflow safe? No hidden gotchas?

If I encrypt with a password:
 
`$ age -p in.tar.xz > out.tar.xz.age`

	
Is it safe to decrypt and pipe straight to tar?

`$ age -d out.tar.xz.age | tar Jxf -`

(By safe, I mean the password isn't leaked to on the command line etc.)

Thanks for any feedback

**Comment by hakavlad** [comment, marked as answer]
The password is requested interactively and does not leak.

Unverified plaintext is not released, it's OK.

Encrypting streams is one of the purposes of age.

Thus, it's safe.

**Reply by older-pack** [reply]
Thank you.
